rooms-quickstart-java/src/main/java/com/communication/rooms/quickstart/App.java

```diff
@@ -134,16 +130,22 @@ public static void updateRoom(String roomId) {
 
     public static void listRooms() {
         try {
-            PagedIterable<CommunicationRoom> rooms = roomsClient.listRooms();
+            Iterable<PagedResponse<CommunicationRoom>> roomPages = roomsClient.listRooms().iterableByPage();
+
+            System.out.println("Listing the first three room IDs in the list of rooms:");
 
             int count = 0;
-            
-            System.out.print("Listing all rooms");
+            for (PagedResponse<CommunicationRoom> page : roomPages) {
+                for (CommunicationRoom room : page.getElements()) {
+                    System.out.println("\n" + room.getRoomId());
+                    count++;
+
+                    if (count >= 3) {
+                        break;
+                    }
+                }
 
-            for (CommunicationRoom room : rooms) {
-                System.out.println("\nFirst two room ID's in the list of rooms: " + room.getRoomId());
-                count++;
-                if (count >= 2) {
+                if (count >= 3) {
```

Comment:
you can remove those lines

Reply:
I think you are counting on rooms. Let print all rooms in the first 2 pages and break out of the loop. You do not need to check inside loop anymore. The inner loop then can be simplified as below.




collection.getElements().forEach(room-> System.out.println(room));